reporting/reporting_sierra_transformer/src/transform.py

```diff
@@ -0,0 +1,165 @@
+        field_name="country",
+        subfields_to_keep="name",
+    )
+
+    # unpack language
+    transformed = unpack(
+        view_record=bib_record,
+        edit_record=transformed,
+        field_name="lang",
+        subfields_to_keep="name",
+    )
+
+    # unpack material types
+    transformed = unpack(
+        view_record=bib_record,
+        edit_record=transformed,
+        field_name="materialType",
+        subfields_to_keep="code",
+    )
+
+    # unpack locations
+    transformed = unpack(
+        view_record=bib_record,
+        edit_record=transformed,
+        field_name="locations",
+        subfields_to_keep=["name", "code"],
+    )
+
+    # unpack orders. note that in the second `unpack()` we view the transformed
+    # record in order to flatten the semi-unpacked order locations
+    transformed = unpack(
+        view_record=bib_record,
+        edit_record=transformed,
+        field_name="orders",
+        subfields_to_keep=["location", "date"],
+    )
+
+    transformed = unpack(
+        view_record=transformed,
+        edit_record=transformed,
+        field_name="orders_location",
+        subfields_to_keep=["name", "code"],
+    )
+
+    # parse the order dates
+    try:
+        transformed["orders_date"] = [
+            parse(date) for date in transformed["orders_date"]
+        ]
+    except KeyError:
+        pass
+
+    # parse publish year
+    transformed = parse_year_int_to_date(
+        view_record=bib_record, edit_record=transformed, field_name="publishYear"
+    )
+
+    return transformed
+
+
+def parse_year_int_to_date(view_record, edit_record, field_name):
+    try:
+        date_to_parse = deepcopy(view_record[field_name])
+        del edit_record[field_name]
+        year_from = datetime(date_to_parse, 1, 1)
+        edit_record[field_name + "_from"] = year_from
+        edit_record[field_name + "_to"] = (
+            year_from + timedelta(days=365) - timedelta(seconds=1)
+        )
+
+    except KeyError:
+        pass
+
+    return edit_record
+
+
+def unpack(view_record, edit_record, field_name, subfields_to_keep):
+    """
+    Parameters
+    ----------
+    view_record : dict
+        original, unchanged record to inspect
+    edit_record : dict
+        a modified/modifiable copy of the original record, to which we will
+        make changes. this is ultimately the unpacked version of the record
+        which we will pass to elasticsearch
+    field_name : str
+        the field whose data we want to unpack
+    subfields to keep : str, list
+        the subfields of the field_name which we want to retain in the
+        modified, flatter version of the data
+    delete : bool
+        should the field be deleted from the transformed record
+
+    Returns
+    -------
+    edit_record : dict
+        the modified record with an unpacked form of the specified field
+    """
+    try:
+        data_to_unpack = deepcopy(view_record[field_name])
+        del edit_record[field_name]
```

Comment:
I think these are the only two lines that can throw a `KeyError`? In general it's good practice to keep the body of your `try` small, so you don't catch and lose unexpected errors.